### PR TITLE
Material/folder name uniqueness case insensitive validation

### PR DESCRIPTION
**Pull request by allenwq**
The upstream just merged my PR, so that we have case insensitive validations now.

Also add more valuations to ensure that materials/folders names are unique among other folders/materials in the same folder.

### CODE REVIEW

**Comment by lowjoel** [comment]
Your PR was supposed to support case insensitive validations. Why are you adding logic to check for uniqueness?

**Comment by allenwq** [comment]
@lowjoel Case insensitive is for uniqueness validation I think ? They are related

**Comment by lowjoel** [comment]
I don't see the point. The schema_validations gem is supposed to make a uniqueness validation without intervention. Why do you need to manually check? Your fix was for making sure it works with a case insensitive index. So your manual checking should no longer be necessary.

**Comment by allenwq** [comment]
Maybe my description is not precise enough, but manually checking was for making sure that a material and a folder cannot have the same name.

Is there any other way to do this except manually checking ?

File: app/models/course/material/folder.rb [hunk 1]
```diff
@@ -27,4 +26,13 @@ def self.after_course_create(course)
   def set_defaults
     self.start_at ||= Time.zone.now
   end
+
+  # TODO: Not threadsafe, consider making all folders as materials
+  # Make sure that folder won't have the same name with other materials in the parent folder
+  def validate_name_is_unique_among_materials
```
Comment (allenwq) [inline app/models/course/material/folder.rb]: @lowjoel I've renamed this to resolve confusion, any better name suggestions?

Comment (lowjoel) [inline app/models/course/material/folder.rb]: Good enough.

File: db/migrate/20151028151248_add_unique_index_to_materials.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToMaterials < ActiveRecord::Migration
+  def change
+    add_index :course_materials, [:name, :folder_id], unique: true, case_sensitive: false
```
Comment (lowjoel) [inline db/migrate/20151028151248_add_unique_index_to_materials.rb]: flip the order of the index, i.e. folder_id, name.

**Comment by allenwq** [comment]
@lowjoel Updated, thanks !

**Comment by allenwq** [comment]
@lowjoel  Thanks a lot!

**Pull request merged**